build.psm1

```diff
@@ -188,6 +190,12 @@ function Start-PSBuild {
     # verify we have all tools in place to do the build
     $precheck = precheck 'dotnet' "Build dependency 'dotnet' not found in PATH. Run Start-PSBootstrap. Also see: https://dotnet.github.io/getting-started/"
 
+    $dotnetIntalledVersion = (dotnet --version)
+    If ( $dotnetIntalledVersion -ne $dotnetRequiredVersion ) {
+        Write-Warning ".Net Core version mismatch: installed version is $dotnetIntalledVersion, required version is $dotnetRequiredVersion.`nRun Start-PSBootstrap and Start-PSBuild -Clean -Restore.`nAlso see: https://dotnet.github.io/getting-started/"
```

Comment:
> now waiting conclusion about auto removing old dotnet version

That can be done with a separate PR. Actually removing the existing `.dotnet` folder from the default location might not be necessary because it seems `dotnet-install.ps1` handles it properly.